packages/intrepid2/src/Shared/Intrepid2_Data.hpp

```diff
@@ -168,20 +168,20 @@ namespace Intrepid2 {
      - BLOCK_PLUS_DIAGONAL: the data varies in this notional dimension and one other, corresponding to a square matrix that has some (possibly trivial) full block, with diagonal entries in the remaining dimensions.  The underlying View will have one dimension corresponding to the two notional dimensions, with extent corresponding to the number of nonzeros in the matrix.
      
   */
-  template<class DataScalar,typename ExecSpaceType = Kokkos::DefaultExecutionSpace>
+  template<class DataScalar,typename DeviceType = Kokkos::DefaultExecutionSpace>
   class Data {
   public:
     using value_type      = DataScalar;
-    using execution_space = ExecSpaceType;
+    using execution_space = typename DeviceType::execution_space;
   private:
     ordinal_type dataRank_;
-    Kokkos::View<DataScalar*,       ExecSpaceType> data1_;  // the rank 1 data that is explicitly stored
-    Kokkos::View<DataScalar**,      ExecSpaceType> data2_;  // the rank 2 data that is explicitly stored
-    Kokkos::View<DataScalar***,     ExecSpaceType> data3_;  // the rank 3 data that is explicitly stored
-    Kokkos::View<DataScalar****,    ExecSpaceType> data4_;  // the rank 4 data that is explicitly stored
-    Kokkos::View<DataScalar*****,   ExecSpaceType> data5_;  // the rank 5 data that is explicitly stored
-    Kokkos::View<DataScalar******,  ExecSpaceType> data6_;  // the rank 6 data that is explicitly stored
-    Kokkos::View<DataScalar*******, ExecSpaceType> data7_;  // the rank 7 data that is explicitly stored
+    Kokkos::View<DataScalar*,       DeviceType> data1_;  // the rank 1 data that is explicitly stored
+    Kokkos::View<DataScalar**,      DeviceType> data2_;  // the rank 2 data that is explicitly stored
+    Kokkos::View<DataScalar***,     DeviceType> data3_;  // the rank 3 data that is explicitly stored
+    Kokkos::View<DataScalar****,    DeviceType> data4_;  // the rank 4 data that is explicitly stored
+    Kokkos::View<DataScalar*****,   DeviceType> data5_;  // the rank 5 data that is explicitly stored
+    Kokkos::View<DataScalar******,  DeviceType> data6_;  // the rank 6 data that is explicitly stored
+    Kokkos::View<DataScalar*******, DeviceType> data7_;  // the rank 7 data that is explicitly stored
```

Comment:
This is not really related comment for this PR. Why do you use this hand version of  DynRankView ?

Reply:
This allows us to use View instead of DynRankView in places where the performance difference might matter.  I instituted this while optimizing for CUDA performance in the sum factorization work.  I don't have hard numbers to show that this matters, but at the time anyway I suspected that it did matter.